Binary files src_checkpoint_01/.DS_Store and src_checkpoint_02/.DS_Store differ
Only in src_checkpoint_01/components: Button.js
diff -rupP src_checkpoint_01/components/DeckScreen/Deck.js src_checkpoint_02/components/DeckScreen/Deck.js
--- src_checkpoint_01/components/DeckScreen/Deck.js	2017-07-05 17:42:07.000000000 -0400
+++ src_checkpoint_02/components/DeckScreen/Deck.js	2017-07-05 17:41:42.000000000 -0400
@@ -13,17 +13,16 @@ class Deck extends Component {
   static displayName = 'Deck';
 
   _review = () => {
-    console.warn("Not implemented");
+    this.props.review();
   }
 
   _addCards = () => {
-    console.warn("Not implemented");
+    this.props.add();
   }
 
   render() {
     return (
       <View style={styles.deckGroup}>
-
         <Button style={styles.deckButton} onPress={this._review}>
           <NormalText>
             {this.props.deck.name}: {this.props.count} cards
@@ -39,7 +38,8 @@ class Deck extends Component {
 }
 
 Deck.propTypes = {
-  deck: React.PropTypes.instanceOf(DeckModel),
+  deck: React.PropTypes.instanceOf(DeckModel).isRequired,
+  add: React.PropTypes.func.isRequired,
   count: React.PropTypes.number
 };
 
diff -rupP src_checkpoint_01/components/DeckScreen/DeckCreation.js src_checkpoint_02/components/DeckScreen/DeckCreation.js
--- src_checkpoint_01/components/DeckScreen/DeckCreation.js	2017-07-04 12:12:01.000000000 -0400
+++ src_checkpoint_02/components/DeckScreen/DeckCreation.js	2017-07-04 13:08:34.000000000 -0400
@@ -13,8 +13,8 @@ class DeckCreation extends Component {
   }
 
   _newDeck = (name) => {
-    console.warn("Not implemented");
     this.setState({ showingNameField: false });
+    this.props.create(name);
   }
 
   _showField = () => {
@@ -31,7 +31,7 @@ class DeckCreation extends Component {
 }
 
 DeckCreation.propTypes = {
-  newDeck: React.PropTypes.func
+  create: React.PropTypes.func.isRequired
 }
 
 export default DeckCreation;
\ No newline at end of file
diff -rupP src_checkpoint_01/components/DeckScreen/index.js src_checkpoint_02/components/DeckScreen/index.js
--- src_checkpoint_01/components/DeckScreen/index.js	2017-07-05 17:42:26.000000000 -0400
+++ src_checkpoint_02/components/DeckScreen/index.js	2017-07-05 17:42:44.000000000 -0400
@@ -8,7 +8,11 @@ import Deck from './Deck';
 import DeckCreation from './DeckCreation';
 
 class DecksScreen extends Component {
-  static displayName = 'DecksScreen'
+  static displayName = 'DecksScreen';
+
+  static navigationOptions = {
+    title: 'All Decks'
+  };
 
   constructor(props) {
     super(props);
@@ -17,6 +21,21 @@ class DecksScreen extends Component {
     }
   }
 
+  _createDeck = () => {
+    console.warn("Data saving not implemented");
+    this.props.navigation.navigate('CardCreation');
+  }
+
+  _addCards = () => {
+    console.warn("Data saving not implemented");
+    this.props.navigation.navigate('CardCreation');
+  }
+
+  _review = () => {
+    console.warn("Actual reviews not implemented");
+    this.props.navigation.navigate('Review');
+  }
+
   _mkDeckViews() {
     if (!this.state.decks) {
       return null;
@@ -27,7 +46,9 @@ class DecksScreen extends Component {
         <Deck
           deck={deck}
           count={deck.cards.length}
-          key={deck.id} />);
+          key={deck.id}
+          add={this._addCards}
+          review={this._review} />);
     });
   }
 
@@ -35,7 +56,7 @@ class DecksScreen extends Component {
     return (
       <View>
         {this._mkDeckViews()}
-        <DeckCreation />
+        <DeckCreation create={this._createDeck}/>
       </View>
     );
   }
diff -rupP src_checkpoint_01/components/Flashcards.js src_checkpoint_02/components/Flashcards.js
--- src_checkpoint_01/components/Flashcards.js	2017-07-05 10:25:21.000000000 -0400
+++ src_checkpoint_02/components/Flashcards.js	2017-07-05 10:25:21.000000000 -0400
@@ -1,27 +1,21 @@
 import React, { Component } from "react";
 import { StyleSheet, View } from "react-native";
+import { StackNavigator } from "react-navigation";
 
-import Heading from "./Header";
+import Logo from "./Header/Logo";
 import DeckScreen from "./DeckScreen";
 import NewCardScreen from "./NewCardScreen";
 import ReviewScreen from "./ReviewScreen";
 
-class Flashcards extends Component {
-  _renderScene() {
-    // return <ReviewScreen />;
-    // return <NewCardScreen />;
-    return <DeckScreen />;
-  }
-  render() {
-    return (
-      <View style={styles.container}>
-        <Heading />
-        {this._renderScene()}
-      </View>
-    );
-  }
-}
+let headerOptions = {
+  headerStyle: { backgroundColor: "#FFFFFF" },
+  headerLeft: <Logo />
+};
 
-const styles = StyleSheet.create({ container: { flex: 1, marginTop: 30 } });
+let navigator = StackNavigator({
+  Home: { screen: DeckScreen, navigationOptions: headerOptions },
+  Review: { screen: ReviewScreen, navigationOptions: headerOptions },
+  CardCreation: { screen: NewCardScreen, navigationOptions: headerOptions }
+});
 
-export default Flashcards;
+export default navigator;
diff -rupP src_checkpoint_01/components/Header/Logo.js src_checkpoint_02/components/Header/Logo.js
--- src_checkpoint_01/components/Header/Logo.js	1969-12-31 19:00:00.000000000 -0500
+++ src_checkpoint_02/components/Header/Logo.js	2017-07-04 13:08:34.000000000 -0400
@@ -0,0 +1,18 @@
+import React, { Component } from 'react';
+import {
+  Image
+} from 'react-native';
+
+import styles from './styles';
+
+class Logo extends Component {
+  static displayName = 'Logo';
+  
+  render() {
+    return (
+        <Image source={require('../../../icon.png')} style={styles.logo}/>
+      );
+  }
+}
+
+export default Logo;
\ No newline at end of file
diff -rupP src_checkpoint_01/components/Header/index.js src_checkpoint_02/components/Header/index.js
--- src_checkpoint_01/components/Header/index.js	2017-07-04 12:12:01.000000000 -0400
+++ src_checkpoint_02/components/Header/index.js	2017-07-04 13:08:34.000000000 -0400
@@ -1,11 +1,11 @@
 import React, { Component } from 'react';
 import {
-  View,
-  Image
+  View
 } from 'react-native';
 
 import styles from './styles';
 import HeadingText from './../HeadingText';
+import Logo from './Logo';
 
 class Header extends Component {
   static displayName = 'Header';
@@ -13,7 +13,7 @@ class Header extends Component {
   render() {
     return (
       <View style={styles.header}>
-        <Image source={require('../../../icon.png')} style={styles.logo}/>
+        <Logo />
         <HeadingText>ZEBRETO</HeadingText>
       </View>
       );
Only in src_checkpoint_01/components: Input.js
Only in src_checkpoint_01/components: LabeledInput.js
diff -rupP src_checkpoint_01/components/NewCardScreen/index.js src_checkpoint_02/components/NewCardScreen/index.js
--- src_checkpoint_01/components/NewCardScreen/index.js	2017-07-04 12:12:01.000000000 -0400
+++ src_checkpoint_02/components/NewCardScreen/index.js	2017-07-04 13:08:34.000000000 -0400
@@ -12,12 +12,18 @@ import NormalText from '../NormalText';
 import colors from './../../styles/colors';
 
 class NewCard extends Component {
+  static navigationOptions = {
+    title: 'Create Card'
+  };
+
+  static initialState = {
+    front: '',
+    back: ''
+  }
+
   constructor(props) {
     super(props);
-    this.state = {
-      font: '',
-      back: ''
-    };
+    this.state = this.initialState;
   }
 
   _handleFront = (text) => {
@@ -29,15 +35,16 @@ class NewCard extends Component {
   }
 
   _createCard = () => {
-    console.warn("Not implemented");
+    console.warn("Data saving not implemented");
+    this.props.navigation.navigate('CardCreation');
   }
 
   _reviewDeck = () => {
-    console.warn("Not implemented");
+    this.props.navigation.navigate('Review');
   }
 
   _doneCreating = () => {
-    console.warn("Not implemented");
+    this.props.navigation.navigate('Home');
   }
 
   render() {
diff -rupP src_checkpoint_01/components/ReviewScreen/index.js src_checkpoint_02/components/ReviewScreen/index.js
--- src_checkpoint_01/components/ReviewScreen/index.js	2017-07-04 12:12:01.000000000 -0400
+++ src_checkpoint_02/components/ReviewScreen/index.js	2017-07-04 13:08:34.000000000 -0400
@@ -13,6 +13,10 @@ class ReviewScreen extends Component {
 
   static displayName = 'ReviewScreen';
 
+  static navigationOptions = {
+    title: 'Review'
+  };
+
   constructor(props) {
     super(props);
     this.state = {
@@ -31,13 +35,15 @@ class ReviewScreen extends Component {
   }
 
   _nextReview = () => {
+    console.warn("Showing next review, but data saving not implemented.");
     this.setState({
       currentReview: this.state.currentReview + 1
     });
   }
 
   _quitReviewing = () => {
-    console.warn("Not implemented");
+    console.warn("Data saving not implemented");
+    this.props.navigation.goBack();
   }
 
   _contents() {
